src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php

```diff
@@ -578,6 +719,72 @@ public function iShouldBeViewingNonArchivalShippingMethods(): void
         // Intentionally left blank
     }
 
+    /**
+     * @Then shipping method :shippingMethod should still have code :code
+     */
+    public function shippingMethodShouldStillHaveCode(ShippingMethodInterface $shippingMethod, string $code): void
+    {
+        Assert::same(
+            $this->responseChecker->getValue($this->client->show(Resources::SHIPPING_METHODS, $shippingMethod->getCode()), 'code'),
+            $code,
+        );
+    }
+
+    /**
+     * @Then I should be notified that amount for :channel channel should not be blank
+     */
+    public function iShouldBeNotifiedThatAmountForChannelShouldNotBeBlank(ChannelInterface $channel): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            sprintf('configuration[%s][amount]: This value should not be blank.', $channel->getCode()),
+        );
+    }
+
+    /**
+     * @Then I should be notified that code needs to contain only specific symbols
+     */
+    public function iShouldBeNotifiedThatCodeNeedsToContainOnlySpecificSymbols(): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            'code: Shipping method code can only be comprised of letters, numbers, dashes and underscores.',
+        );
+    }
+
+    /**
+     * @Then I should be notified that shipping charge for :channel channel cannot be lower than 0
+     */
+    public function iShouldBeNotifiedThatShippingChargeForChannelCannotBeLowerThan0(ChannelInterface $channel): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            sprintf('configuration[%s][amount]: Shipping charge cannot be lower than 0.', $channel->getCode()),
+        );
+    }
+
+    /**
+     * @Then I should be notified that the weight rule has an invalid configuration
+     */
+    public function iShouldBeNotifiedThatTheWeightRuleHasAnInvalidConfiguration(): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            'rules[0].configuration[weight]: This value should be of type numeric.',
```

Comment:
```suggestion
            'configuration[weight]: This value should be of type numeric.',
```
As is it cannot be reused in scenarios with multiple rules